### PR TITLE
[TS Client] Test message size

### PR DESCRIPTION
**Pull request by BrennanConroy**
As well as adding tests, reduced invoke, send, and stream message sizes by 15 bytes when no streaming parameters are present.

### CODE REVIEW

**Comment by anurse** [comment]
Wakey wakey.

**Comment by anurse** [comment]
Let's just merge this for JSON.

File: src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import { NullLogger } from "../src/Loggers";
+import { Subject } from "../src/Subject";
+import { VerifyLogger } from "./Common";
+import { TestConnection } from "./TestConnection";
+import { delayUntil, registerUnhandledRejectionHandler } from "./Utils";
+
+registerUnhandledRejectionHandler();
+
+function createHubConnection(connection: IConnection, logger?: ILogger | null, protocol?: IHubProtocol | null) {
+    return HubConnection.create(connection, logger || NullLogger.instance, protocol || new JsonHubProtocol());
```
Comment (halter73) [inline src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts]: Are the TS tests for MessagePack? It looks like these are only testing JsonHubProtocol.

File: src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            try {
+                const subject = new Subject();
+                hubConnection.stream("target", 1, subject);
+                subject.complete();
+
+                await delayUntil(1000, () => connection.sentData.length === 2);
+
+                // Verify the message is sent
+                expect(connection.sentData.length).toBe(2);
+                expect((connection.sentData[1] as string).length).toEqual(29);
```
Comment (halter73) [inline src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts]: Can we verify the message type is what we expect before verifying the length?

File: src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts (hunk 1 above)
Comment (BrennanConroy) [reply src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts]: No tests for MessagePack, that's what we were talking about Friday when we decided to just get the PR in.

File: src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts (hunk 2 above)
Comment (BrennanConroy) [reply src/SignalR/clients/ts/signalr/tests/MessageSize.test.ts]: Done

**Review by halter73 — APPROVED** [review]

**Pull request merged**